torch/csrc/jit/codegen/cuda/lower_thread_predicate.cpp

```diff
@@ -43,21 +41,30 @@ kir::Val* getPredicatePerParallelType(
         pt);
     const auto& source = source_it->second;
     TORCH_INTERNAL_ASSERT(!source.empty(), "No predicate source found");
+    kir::Val* pred = ir_builder.trueVal();
     for (auto src : source) {
       auto flag_name = kir::GridReduction::getPredicateFlagName(src);
       auto src_pred =
           ir_builder.create<kir::NamedScalar>(flag_name, DataType::Bool);
       pred = ir_builder.andExpr(pred, src_pred);
     }
-    return pred;
+    // Make it Bool for convenience
+    if (pred->isA<kir::NamedScalar>()) {
```

Comment:
This looks like implicitly casting a named scalar directly to bool. Just curious do we have new kinds of named scalar or this is just checking if the thread idx's are 0?

It'd look nicer to me to have some comments here especially if it's the latter case.

Reply:
Expanded the comment.